src/ol/source/stamensource.js

```diff
@@ -88,36 +88,78 @@ ol.source.StamenProviderConfig = {
  */
 ol.source.Stamen = function(options) {
 
-  var i = options.layer.indexOf('-');
-  var provider = i == -1 ? options.layer : options.layer.slice(0, i);
-  goog.asserts.assert(provider in ol.source.StamenProviderConfig,
-      'known provider configured');
-  var providerConfig = ol.source.StamenProviderConfig[provider];
+  /**
+   * url
+   * @type {string}
+   * @private
+   */
 
-  goog.asserts.assert(options.layer in ol.source.StamenLayerConfig,
-      'known layer configured');
-  var layerConfig = ol.source.StamenLayerConfig[options.layer];
+  this.rootUrl_ = goog.isDef(options.url) ? options.url :
+      'https://stamen-tiles-{a-d}.a.ssl.fastly.net/';
```

Comment:
Once something is stable, we cannot make it unstable again. So please just proceed as suggested above, and update the documentation of the `url` option in [`olx.js`](https://github.com/SDaron/ol3/blob/stamen-source/externs/olx.js#L4555-L4560). Also, a mention of this change in the [upgrade notes](https://github.com/SDaron/ol3/blob/stamen-source/changelog/upgrade-notes.md) would be good.